src/ppx/xr_component.cpp

```diff
@@ -48,6 +48,43 @@ static XrBool32 XrDebugUtilsMessengerCallback(XrDebugUtilsMessageSeverityFlagsEX
     return XR_FALSE; /* OpenXR spec suggests always returning false. */
 }
 
+template <int N>
+void CharArrayCopy(char (&dst)[N], const char* src)
+{
+    strncpy(dst, src, N);
+    // Force the last byte to be '\0'
+    dst[N - 1] = '\0';
```

Comment:
would it be better to have src as string, and assert `src.length() <= N`?

Reply:
`assert(strlen() < N)` should also work.
There's not much reason to use assert here, a truncated app name or action name is not fatal (and probably won't happen anyway).

Reply:
Updated to string_view, and added assert.